### PR TITLE
fixing bug with user claims.  PR queue number:   1

### PR DESCRIPTION
**Pull request by Alona-Kirichenko**
- deleting userLevelConext and creating userLevel store with mobx to store usel level
- updating tests
- moving context providers to navigation file to wrap screens directly there and keep App file cleaner

### CODE REVIEW

File: src/utilily/firebaseTypes.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { User } from "./types";
+
+export type UserClames = {
```
Comment (covetous) [inline src/utilily/firebaseTypes.ts:3]: UserClaims*

File: src/utilily/firebaseTypes.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  admin?: boolean;
+  superadmin?: boolean;
+  user?: boolean;
+
+  aud?: string;
```
Comment (covetous) [inline src/utilily/firebaseTypes.ts:8]: Kommentar om varför aud, exp, iat, iss, och sub är med i modellen.

File: src/utilily/firebaseTypes.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  auth_time?: number;
+  email?: string;
+  email_verified?: boolean;
+  exp?: number;
+  firebase?: { identities: { email: [string] }; sign_in_provider: string };
+  iat?: number;
+  iss?: string;
+  sub?: string;
+  user_id?: User["id"];
```
Comment (Mattan1999) [inline src/utilily/firebaseTypes.ts:17]: I'd say we only need our own custom claims here.

Comment (Alona-Kirichenko) [reply src/utilily/firebaseTypes.ts:17]: We need to have type of the object we are receiving. Right now we are receiving claims object that looks like this
{
  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;
  aud?: string;
  auth_time?: number;
  email?: string;
  email_verified?: boolean;
  exp?: number;
  firebase?: { identities: { email: [string] }; sign_in_provider: string };
  iat?: number;
  iss?: string;
  sub?: string;
  user_id?: User["id"];
};  


I don't think it's a good idea to say that claims of type

 {  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;  }

And then press in it object that looks like this.

  {
  admin?: boolean;
  superadmin?: boolean;
  user?: boolean;
  aud?: string;
  auth_time?: number;
  email?: string;
  email_verified?: boolean;
  exp?: number;
  firebase?: { identities: { email: [string] }; sign_in_provider: string };
  iat?: number;
  iss?: string;
  sub?: string;
  user_id?: User["id"];
};

File: src/screens/ActivityCard.tsx [hunk 4]
```diff
@@ -37,7 +37,7 @@ export function ActivityCard({ route, navigation }: Props) {
   const { activityHasChangedID } = useCreateActivityFunction();
   const { setCleanUpSearchBarComponent } = useAdminGalleryFunction();
   const { getImageForActivity } = useActivityImages();
-  const userLevel = useUserLevelCheckFunction();
+  const userLevel = userLevelStore.userLevel;
```
[the remark this answers is not in the document]
Comment (Alona-Kirichenko) [reply src/screens/ActivityCard.tsx:40]: How ? show me.
here is userLevelStore 
import { makeObservable, observable, action } from "mobx";
import { Role } from "../utilily/enums";

class UserLevelStore {
  userLevel: Role | undefined = undefined;

  constructor() {
    makeObservable(this, {
      userLevel: observable,
      setUserLevel: action,
    });
  }

  setUserLevel(level: Role | undefined) {
    this.userLevel = level;
  }
}

const userLevelStore = new UserLevelStore();
export default userLevelStore;

File: src/hooks/useAuthStateListener.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+        userLevelStore.setUserLevel(
+          userIdToken?.claims?.admin
+            ? Role.admin
+            : userIdToken?.claims?.superadmin
+              ? Role.superadmin
+              : userIdToken?.claims?.user
+                ? Role.user
+                : undefined,
+        );
```
[the remark this answers is not in the document]
Comment (Alona-Kirichenko) [reply src/hooks/useAuthStateListener.ts:33]: yes, would be more nice with destructure it
 with const { userLevel } = userLevelStore works good but with  setUserLevel  doesn't work. 
I was trying to find why but don't want to put too much time on. I think it fine userLevelStore.setUserLevel for now

[the remark this answers is not in the document]
Comment (Alona-Kirichenko) [reply src/hooks/useAuthStateListener.ts:33]: or you can take a look. I didn't work with mobx before

File: src/utilily/firebaseTypes.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { User } from "./types";
+
+// I wrote this type by looking at what data we are receiving. We should look over it and maybe change what we want to receive
```
Comment (covetous) [inline src/utilily/firebaseTypes.ts:3]:
```suggestion
// This represents the entire object that we receive from firebase. 
// Not all objects in the model is used, but felt better to handle them, then to ignore them.
```

**Review by Mattan1999 — APPROVED** [review]

**Pull request merged**